File: 040-Find Count of Magical Numbers from 1 to N/MagicalNumbers.py
def is_magical(num):
    binary = bin(num)[2:]  # Remove '0b' prefix
    transformed = ''.join('1' if b == '0' else '2' for b in binary)
    digit_sum = sum(int(digit) for digit in transformed)
    return digit_sum % 2 == 1

def count_magical_numbers(N):
    return sum(1 for i in range(1, N + 1) if is_magical(i))

# Example
N = int(input("Enter a Number: "))
print(count_magical_numbers(N))

File: 040-Find Count of Magical Numbers from 1 to N/MagicalNumbers.cpp
#include<iostream>
#include<string>
using namespace std;

bool isMagical(int num){
    string binary = "";
    while(num > 0){
        binary = (char)((num % 2) + '0') + binary;
        num /= 2; 
    }

    int sum = 0;
    for(char ch : binary){
        if(ch == '0'){
            sum += 1;
        }
        else if(ch == '1'){
            sum += 2;
        }
    }
    return sum % 2 == 1;
}

int countMagicalNumbers(int N){
    int count = 0;
    for(int i = 1; i <= N; i++){
        if(isMagical(i)){
            count++;
        }
    }
    return count;
}
int main(){
    int N;
    cout<<"Enter a Number: ";
    cin >> N;
    cout<< countMagicalNumbers(N) <<endl;
    return 0;
}